api/v1alpha1/stage_status_types.go

```diff
@@ -0,0 +1,79 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StageStatusPhase defines the observed stage phase
+// +kubebuilder:validation:Enum={Progressing,Succeeded,Failed}
+type StageStatusPhase string
+
+const (
+	PhaseProgressing StageStatusPhase = "Progressing"
+	PhaseSucceeded   StageStatusPhase = "Succeeded"
+	PhaseFailed      StageStatusPhase = "Failed"
+)
+
+// StageStatus defines the observed stage status
+type StageStatus struct {
+	Name       string           `json:"name"`
+	Phase      StageStatusPhase `json:"stage,omitempty"`
+	Message    string           `json:"message,omitempty"`
+	StartedAt  *metav1.Time     `json:"startedAt,omitempty"`
+	FinishedAt *metav1.Time     `json:"finishedAt,omitempty"`
+}
+
+// GetStageStatus returns a pointer to the Status.Stages slice
+func (in *ProgressiveRollout) GetStageStatus() *[]StageStatus {
```

Comment:
this should be a slice, not a pointer to a slice

Reply:
(generally speaking, there should not be any pointer to slices, it's already a pointer)